server-spi-private/src/main/java/org/keycloak/broker/provider/util/SimpleHttp.java

```diff
@@ -141,42 +147,25 @@ public String asString() throws IOException {
                 connection.setRequestProperty("Content-Length", String.valueOf(data.length()));
 
                 os = connection.getOutputStream();
-		
-		logger.debug("Request string for url (" + this.url + "): " + data);
-		    
+
+                logger.debugf("Request body for url (%s): %s ", this.url, data);
+
                 os.write(data.getBytes());
             } else {
                 connection.setDoOutput(false);
             }
 
             String ce = connection.getHeaderField("Content-Encoding");
-		try {
-			java.util.Map<String, java.util.List<String>> hdrs = connection.getHeaderFields();
-			java.util.Set<String> hdrKeys = hdrs.keySet();
-		    	for (String k : hdrKeys) {
-		      		logger.debug("Response Header: " + k + "  Value: " + hdrs.get(k));
-			}
-            		is = connection.getInputStream();
-		} catch (IOException ioerrexcp) {
-			InputStream erris = null;
-			try {
-				erris = connection.getErrorStream();
-				java.util.Scanner errscn = new java.util.Scanner(erris).useDelimiter("\\A");
-    				String errbody = errscn.hasNext() ? errscn.next() : "";
-				logger.error("IOException when reading connection input stream: " + errbody + "; " + ioerrexcp);
-				throw ioerrexcp;
-			} finally {
-				if (erris != null) {
-			                try {
-                 			   erris.close();
-                			} catch (IOException e) {
-                			}
-				}
-			}
-		}
+            try {
+                logResponseHeaders(connection);
+                is = connection.getInputStream();
+            } catch (IOException ioerrexcp) {
+                return logErrorResponse(connection, ioerrexcp);
+            }
+
             if ("gzip".equals(ce)) {
-              is = new GZIPInputStream(is);
-	          }
+                is = new GZIPInputStream(is);
```

Comment:
Well... this whole file is so wrongly formatted.I probably should run autoformat on it.